Implementation for Engagement_Presets and Network_Locations API

jpbowie opened the pull request:
Added support to the v2 API for Engagement_Presets and Network_Locations

StefanFl reviewed (changes requested):
Thanks for your submission @jpbowie. With some changes for the permissions it will be good for merging. 

Just out of interest, I never saw why the EngagementPresets are needed. What is your use case for them?

jpbowie commented:
I simplified the engagement presets permissions implementation and removed the network locations implementation in favor of the model permissions as requested.  I removed the  engagement presets validation as it was simply replacing a null scope with an empty space to prevent errors, the user can do that in the api.

I'm thinking about using the engagement presets api with some automation so when a new engagement is created based on a preset, specific tests within the preset can be performed automatically and the results can then be uploaded as part of the engagement.  However, for this to be possible, the tests that are part of the preset need to be available and I didn't find an existing api that would allow me to determine that.

StefanFl commented:
There is currently no functionality implemented concerning `Engagement_Presets`. You can save new ones and edit them, but that's it.

jpbowie commented:
There appears to be enough functionality to implement what I was thinking (except for the API support).  I have been able to Add/Edit/Delete an Engagement_Preset, assign an Engagement_Preset to an existing or new engagement (both in the UI and the engagement API), and an Engagement_Preset panel appears on the engagement screen with details when an Engagement_Preset has been assigned to the engagement.
